### PR TITLE
add stylint

### PR DESCRIPTION
**Pull request by maxlang**
Open to suggestions for `.stylintrc` settings.

### CODE REVIEW

**Comment by maxlang** [comment]
Remake of #2780

**Comment by maxlang** [comment]
@tamird @mrtracy @BramGruneir any comments?

I'd like to settle on this since if it changes I'll have to rewrite the Stylus I'm working on now.

File: ui/ts/components/navbar.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ module Components {
   export module NavigationBar {
     import MithrilVirtualElement = _mithril.MithrilVirtualElement;
     export interface Target {
-      view: string | MithrilVirtualElement | MithrilVirtualElement[];
+      view: string | MithrilVirtualElement[];
```
Comment (mrtracy) [inline ui/ts/components/navbar.ts]: These changes appear to be unrelated; they appear to undo a previous improvement, actually.

Comment (maxlang) [inline ui/ts/components/navbar.ts]: Ugh, how did that happen.

**Comment by mrtracy** [comment]
LGTM, no problem with the changes made.  Tweaking the linter settings later should be relatively easy.
